syntax = "proto3";

package support_bot;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  // Sends another greeting
  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

service MetaHandler {
  // Fetch avatar mxc URL
  rpc FetchAvatarURL (AvatarURLRequest) returns (AvatarURLReply) {}
}

message AvatarURLRequest{
  string user_id = 1;
}

message AvatarURLReply {
  string avatar_url = 1;
  string error_code = 2;
}

